opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -198,38 +217,75 @@ public List<DateFormatter> getTimeNamedFormatters() {
     return getFormatList().stream()
         .filter(formatString -> {
           FormatNames namedFormat = FormatNames.forName(formatString);
-          return SUPPORTED_NAMED_TIME_FORMATS.contains(namedFormat);
+          return namedFormat == null ? false : SUPPORTED_NAMED_TIME_FORMATS.contains(namedFormat);
         })
         .map(DateFormatter::forPattern).collect(Collectors.toList());
   }
 
-  private ExprCoreType getExprTypeFromFormatString(String formatString) {
-    if (formatString.isEmpty()) {
-      // FOLLOW-UP: check the default formatter - and set it here instead
-      // of assuming that the default is always a timestamp
-      return TIMESTAMP;
+  private ExprCoreType getExprTypeFromCustomFormatString(List<DateFormatter> formatters) {
+    // Characters from
+    // https://opensearch.org/docs/latest/field-types/supported-field-types/date/#built-in-formats
+    Pattern timeFormatChars = Pattern.compile("[HmsSZ]");
```

Comment:
`h` for 12 hour clock?
OpenSearch docs say almost nothing about custom formats, but [elastic docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html#custom-date-formats) refer to [java docs](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html). I think we should support all java pattern letters.

Reply:
Updated checking to use java DateTimeFormatter